### PR TITLE
[pegtl] Update to 3.0.0 release

### PR DESCRIPTION
**Pull request by wravery**
**Describe the pull request**

- What does your PR fix?

https://github.com/taocpp/PEGTL just released version 3.0.0. This syncs the vcpkg port with that release tag.

- Which triplets are supported/not supported? Have you updated the CI baseline?

This is a header-only library with no dependencies outside of the C++17 standard library. It does require `std::filesystem`, but it has fallback logic in CMake to auto-detect the `std::experimental::filesystem` implementation or additional linked library requirements that are required on older toolchains.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Yes.

### CODE REVIEW

**Comment by JackBoosY** [comment]
```
/mnt/vcpkg-ci/installed/x64-linux/include/tao/pegtl/read_input.hpp:7:10: fatal error: filesystem: No such file or directory
 #include <filesystem>
          ^~~~~~~~~~~~
compilation terminated.
```
See https://stackoverflow.com/questions/39231363/fatal-error-filesystem-no-such-file-or-directory.
Can you please fix it in this PR?

Thanks.

**Comment by JackBoosY** [comment]
No achives were found in the results, will rerun the pipeline test.

**Comment by wravery** [comment]
Trouble is PEGTL has dropped support for those compiler versions (so cppgraphqlgen, which depends on PEGTL did as well) since they don't fully implement C++17. To fix it would mean patching PEGTL in a way that I don't think the upstream maintainer would want to take. Is it better to patch the port (and see about upstreaming the patch later), or is there a process for deprecating these older compiler toolchains for a given port?

**Comment by JackBoosY** [comment]
@wravery We want to use the patch method accepted by upstream, do you have any ideas?

**Comment by wravery** [comment]
We've sort of been discussing this in https://github.com/taocpp/PEGTL/issues/216, I have an idea that might work. Let me see if the project owners like it.

**Comment by wravery** [comment]
Accidentally force pushed the reset instead of the replacement.

**Review by ras0219 — APPROVED** [review]

File: ports/pegtl/CONTROL [hunk 1]
```diff
@@ -1,4 +1,4 @@
 Source: pegtl
-Version: 3.0.0-pre-70971f4
+Version: 3.0.0
```
Comment (JackBoosY) [inline ports/pegtl/CONTROL:2]:
```suggestion
Version: 20201207
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**